### PR TITLE
Gpu test workflow

### PR DESCRIPTION
**Pull request by ninginthecloud**
Please read through our [contribution guide](https://github.com/pytorch-labs/torcheval/blob/main/CONTRIBUTING.md) prior to creating your pull request.

Summary:
Update workflow with gpu test. 

Test plan:

Fixes #{issue number}

### CODE REVIEW

File: examples/distributed_example.py [hunk 1]
```diff
@@ -4,6 +4,16 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
+"""
```
Comment (edward-io) [inline examples/distributed_example.py:7]: did you accidentally check in this file?

File: .github/workflows/unit_test.yaml [hunk 2]
```diff
@@ -36,3 +36,72 @@ jobs:
...
+          sudo yum remove -y cuda-drivers || true
+          sudo yum remove -y "*nvidia*" || true
+      - name: Install nvidia driver, nvidia-docker runtime, set GPU_FLAG
+        uses: nick-fields/retry@71062288b76e2b6214ebde0e673ce0de1755740a
+        with:
+          timeout_minutes: 10
+          max_attempts: 3
+          command: |
+            set -ex
+            bash .github/scripts/install_nvidia_utils_linux.sh
```
Comment (edward-io) [inline .github/workflows/unit_test.yaml:72]: this file needs to be copied over

File: examples/distributed_example.py (hunk 1 above)
Comment (ninginthecloud) [reply examples/distributed_example.py:7]: yep. sorry, my bad. Let me clean this file.

File: .github/workflows/unit_test.yaml (hunk 2 above)
Comment (ninginthecloud) [reply .github/workflows/unit_test.yaml:72]: yep. just copied in next version.

File: .github/workflows/unit_test.yaml [hunk 3]
```diff
@@ -37,3 +37,39 @@ jobs:
...
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v2
+      - name: Setup conda env
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniconda-version: "latest"
+          activate-environment: test
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
```
Comment (ananthsub) [inline .github/workflows/unit_test.yaml:56]: should we do what torchsnapshot does for gpu testing here? 

https://github.com/pytorch/torchsnapshot/blob/a5e53e21527dda7f4491d4915fe7e6897fdee4ca/.github/workflows/run_tests.yaml#L56-L70

File: .github/scripts/install_nvidia_utils_linux.sh [hunk 4]
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
```
Comment (ananthsub) [inline .github/scripts/install_nvidia_utils_linux.sh:1]: do we still need this?

File: .github/workflows/unit_test.yaml (hunk 3 above)
Comment (ninginthecloud) [reply .github/workflows/unit_test.yaml:72]: Great suggestion! This makes more clean! 

> should we do what torchsnapshot does for gpu testing here?
> 
> https://github.com/pytorch/torchsnapshot/blob/a5e53e21527dda7f4491d4915fe7e6897fdee4ca/.github/workflows/run_tests.yaml#L56-L70

Great suggestion! This makes more clean!

**Pull request closed without merging**